Sources/Workspace/Workspace.swift

```diff
@@ -1029,51 +1030,52 @@ extension Workspace {
-            let manifestsMap: [String: Manifest] = Dictionary(uniqueKeysWithValues:
-                root.manifests.map({ (PackageReference.computeIdentity(packageURL: $0.url), $0) }) +
-                dependencies.map({ (PackageReference.computeIdentity(packageURL: $0.manifest.url), $0.manifest) }))
+            let manifestsMap: [PackageIdentity: Manifest] = Dictionary(uniqueKeysWithValues:
+                root.manifests.map { (PackageIdentity($0.url), $0) } +
+                    self.dependencies.map { (PackageIdentity($0.manifest.url), $0.manifest) }
+            )
 
             var inputIdentities: Set<PackageReference> = []
-            let inputNodes: [GraphLoadingNode] = root.manifests.map({ manifest in
-                let identity = PackageReference.computeIdentity(packageURL: manifest.url)
-                let package = PackageReference(identity: identity, path: manifest.url, kind: manifest.packageKind)
+            let inputNodes: [GraphLoadingNode] = self.root.manifests.map { manifest in
+                let identity = PackageIdentity(manifest.url)
+                let package = PackageReference(identity: identity.description, path: manifest.url, kind: manifest.packageKind)
                 inputIdentities.insert(package)
                 let node = GraphLoadingNode(manifest: manifest, productFilter: .everything)
                 return node
-            }) + root.dependencies.compactMap({ dependency in
+            } + self.root.dependencies.compactMap { dependency in
                 let url = workspace.config.mirrors.effectiveURL(forURL: dependency.url)
-                let identity = PackageReference.computeIdentity(packageURL: url)
-                let package = PackageReference(identity: identity, path: url)
+                let identity = PackageIdentity(url)
+                let package = PackageReference(identity: identity.description, path: url)
                 inputIdentities.insert(package)
                 guard let manifest = manifestsMap[identity] else { return nil }
                 let node = GraphLoadingNode(manifest: manifest, productFilter: dependency.productFilter)
                 return node
-            })
+            }
 
             var requiredIdentities: Set<PackageReference> = []
             _ = transitiveClosure(inputNodes) { node in
-                return node.manifest.dependenciesRequired(for: node.productFilter).compactMap({ dependency in
+                node.manifest.dependenciesRequired(for: node.productFilter).compactMap { dependency in
                     let url = workspace.config.mirrors.effectiveURL(forURL: dependency.url)
-                    let identity = PackageReference.computeIdentity(packageURL: url)
-                    let package = PackageReference(identity: identity, path: url)
+                    let identity = PackageIdentity(url)
+                    let package = PackageReference(identity: identity.description, path: url)
```

Comment:
can we use the actual identity instead of its description?

Reply:
You've correctly picked up on one of the more awkward details of the transition. `manifestMap` was changed to be keyed on `PackageIdentity` values, but ([as discussed above](https://github.com/apple/swift-package-manager/pull/3032/files/96598b51d49ffd617f89ceacbbfb014e191504f3#r520148388)), `PackageReference` doesn't (for now) expose its `identity` as an `PackageIdentity`.

We could make changes here to get around the conceit of calling `description` directly here, but those would likely go away in the next phase (See [this comment](https://github.com/apple/swift-package-manager/pull/3032/files/96598b51d49ffd617f89ceacbbfb014e191504f3#r520746751)).

Reply:
I see. my concern in the two step approach is ending up with a mixed bag, but lets see what @neonichu thinks.

Reply:
I gave this another pass and came up with 5981043bcd332ec3d61336002609c2602454ded3. Let me know if that feels better to you.